flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/deployment/AbstractFlinkResourceReconciler.java

```diff
@@ -97,6 +100,8 @@ public final void reconcile(CR cr, Context<?> ctx) throws Exception {
             return;
         }
 
+        setOwnerReference(cr, deployConfig);
+
```

Comment:
I think we should call this method from inside the `deploy` method otherwise the ownerreference might not be set correctly during rollbacks, restarts etc

Reply:
You are right, it might be have too broad impact to call this method from there :)